### PR TITLE
Deny wrong pull

### PR DESCRIPTION
**Pull request by M4RKUS-11111**
Add check if head is equal to base

### CODE REVIEW

**Review by 6543 — APPROVED** [review]
I'm not sure about the error message - but the rest is fine ;D

File: routers/api/v1/repo/pull.go [hunk 1]
```diff
@@ -284,6 +284,12 @@ func CreatePullRequest(ctx *context.APIContext, form api.CreatePullRequestOption
 	//   "422":
 	//     "$ref": "#/responses/validationError"
 
+	if form.Head == form.Base {
+		ctx.Error(http.StatusUnprocessableEntity, "BaseHeadSame",
+			"Invalid PullRequest: No changes presented. ( Head is equal to Base )")
```
Comment (6543) [inline routers/api/v1/repo/pull.go]:
```suggestion
			"Head and Base are not allowed to be the same")
```

Comment (6543) [reply routers/api/v1/repo/pull.go]: @techknowlogick do you have a better one :point_up: ?

Comment (zeripath) [reply routers/api/v1/repo/pull.go]:
```suggestion
			"Invalid PullRequest: There are no changes between the head and the base")
```

**Review by lafriks — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by lafriks** [comment]
please send backports to release/v1.12 and release/v1.13 branches

**Pull request merged**